StackOnUList/MyStack.cpp

```diff
@@ -0,0 +1,93 @@
+#include "StdAfx.h"
+#include "MyStack.h"
+
+
+MyStack::MyStack(void): s_size(0), s_top(NULL)
+{
+}
+
+MyStack::MyStack(const MyStack &base): s_size(0), s_top(NULL)
+{
+    if (base.s_top == NULL)
+        return;
+
+    copy_stack(base.s_top);
+}
+
+MyStack::~MyStack(void)
+{
+    while (!empty())
+        pop();
+}
+
+MyStack& MyStack::operator=(const MyStack &base)
+{
+    if (base.s_top == NULL)
+        return *this;
+
+    copy_stack(base.s_top);
+
+    return *this;
+}
+
+void MyStack::push(Data val)
+{
+    UList *p = new UList;
+    p->l_data = val;
+    p->l_next = s_top;
+
+    s_top = p;
+    s_size++;
+}
+
+
+bool MyStack::pop()
+{
+    if (s_top == NULL)
+        return false;
+
+    UList *next_top = s_top->l_next;
+    
+    delete s_top;
+
+    s_top = next_top;
+    s_size--;
+
+    return true;
+}
+
+void MyStack::swap(MyStack &sec_stack)
+{
+    size_t tmp_s_size = s_size;
+    UList *tmp_s_top = s_top;
+
+    s_size = sec_stack.s_size;
+    s_top = sec_stack.s_top;
+
+    sec_stack.s_size = tmp_s_size;
+    sec_stack.s_top = tmp_s_top;
+}
```

Comment:
Using std::swap would be more readable. 2 lines instead of 6 lines, essentially doing the same thing.